test.nix

```diff
@@ -174,9 +174,37 @@ let
       git-dep = naersk.buildPackage {
         doCheck = true;
         src = ./test/git-dep;
+      };
+
+      git-dep-by-branch = naersk.buildPackage {
```

Comment:
This test was not present previously in here (even though the directory already existed); not sure why, so I've brought it back.